AutoMapper.AspNetCore.OData.EFCore/FilterHelper.cs

```diff
@@ -29,6 +29,9 @@ public FilterHelper(IDictionary<string, ParameterExpression> parameters, Type un
             this.edmModel = context.Model;
         }
 
+        private const string DollarThis = "$this";
+        private const string DollarIt = "$it";
```

Comment:
`filterClause.RangeVariable.Name` might be be better than hardcoding $it here - ok to move at least for this PR I think.

Reply:
We're replacing the `$this` literal with the `$it` literal. The `filterClause.RangeVariable.Name` is equal to `$this` which is what we replace with `$it`. If `filterClause.RangeVariable.Name` is not equal to `$this` then we use `filterClause.RangeVariable.Name` and not the hardcoded `$it`.

Reply:
Not what I meant - we don't  a case where the range variable [here](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/6a5e4ce73a5e4f0c71b4e675facc54e7f7a1f0ac/AutoMapper.AspNetCore.OData.EFCore/LinqExtensions.cs#L589) is not `$it`.

I'm referring to a scenario where Microsoft.AspNetCore.OData changes `$it` to something different.